pkg/controller/nodeipam/ipam/cidrset/cidr_set_test.go

```diff
@@ -341,6 +343,77 @@ func TestCIDRSet_AllocationOccupied(t *testing.T) {
 	}
 }
 
+func TestDoubleOccupyRelease(t *testing.T) {
+	clusterCIDRStr := "10.42.0.0/16"
+	operations := []struct {
+		cidrStr     string
+		occupy      bool
+		numOccupied int
+	}{
+		{
+			cidrStr:     "10.42.5.0/24",
+			occupy:      true,
+			numOccupied: 1,
+		},
+		{
+			cidrStr:     "10.42.9.0/24",
+			occupy:      true,
+			numOccupied: 2,
+		},
+		{
+			cidrStr:     "10.42.8.0/22",
+			occupy:      true,
+			numOccupied: 5,
+		},
+		{
+			cidrStr:     "10.42.9.0/24",
+			occupy:      true,
+			numOccupied: 5,
+		},
+		{
+			cidrStr:     "10.42.9.0/24",
+			occupy:      false,
+			numOccupied: 4,
+		},
+		{
+			cidrStr:     "10.42.9.0/24",
+			occupy:      false,
+			numOccupied: 4,
+		},
+	}
+	numAllocatable24s := (1 << 8) - 4
+
+	_, clusterCIDR, _ := net.ParseCIDR(clusterCIDRStr)
+	a, err := NewCIDRSet(clusterCIDR, 24)
+	if err != nil {
+		t.Fatalf("Error allocating CIDRSet")
+	}
+
+	for _, op := range operations {
+		_, cidr, _ := net.ParseCIDR(op.cidrStr)
+		if op.occupy {
+			a.Occupy(cidr)
+		} else {
+			a.Release(cidr)
+		}
+		if a.allocatedCIDRs != op.numOccupied {
+			t.Fatalf("Expected %d occupied CIDRS, got %d", op.numOccupied, a.allocatedCIDRs)
+		}
+	}
+
+	for i := 0; i < numAllocatable24s; i++ {
```

Comment:
add comment here explaining that we are going to allocate the rest of the subnets, to fill the pool